tests/data_files/Makefile

```diff
@@ -439,7 +439,7 @@ parse_input/server5-directoryname.crt.der: server5.key
 	$(OPENSSL) req -x509 -outform der -new -subj "/C=UK/O=Mbed TLS/CN=Mbed TLS directoryName SAN" -set_serial 77 -config $(test_ca_config_file) -extensions directory_name_san -days 3650 -sha256 -key server5.key -out $@
 
 parse_input/server5-two-directorynames.crt.der: server5.key
-	$(OPENSSL) req -x509 -outform der -new -subj "/C=UK/O=Mbed TLS/CN=Mbed TLS directoryName SAN" -set_serial 77 -config $(test_ca_config_file) -extensions two_directorynames -days 3650 -sha256 -key server5.key -out $@
+	$(OPENSSL) req -x509 -outform der -new -subj "/C=UK/O=Mbed TLS/CN=Mbed TLS directoryName SAN" -set_serial 77 -config $(test_ca_config_file) -extensions bad_second_directory_name_san -days 3650 -sha256 -key server5.key -out $@
```

Comment:
I don't understand this change. Why isn't the `two_directorynames` section the one we want to generate the `two-directorynames` file?

Looking at the content of [`test-ca.opensslconf`](https://github.com/Mbed-TLS/mbedtls/blob/dfc2e26ddf2960742e4041871bfa0cc6f624790c/tests/data_files/test-ca.opensslconf#L111), the line
```
subjectAltName=dirName:dirname_sect, dirName:dirname_sect_bad
```
which is in the section `bad_second_directory_name_san` does seem plausible for what `server5-two-directorynames.crt` is about. That section is currently unused. But after this change the section `two_directorynames` will be unused as well, which doesn't look right. Is `O=MALFORM_ME` needed? (I don't understand why we'd want it, but I don't fully understand what the file is testing.) Should we remove the section containing `O=MALFORM_ME`?

[I left a comment on the original PR](https://github.com/Mbed-TLS/mbedtls/pull/6900/files#r1244796928), hoping the author or reviewers can answer.

Reply:
As noted in the original PR, `bad_second_directory_name_san` is an unused leftover version that should be removed. `two_directorynames` is the valid extension used to set up a basic, working file for two further malformations done in `server5-directoryname-seq-malformed.crt.der` and `server5-second-directoryname-oid-malformed.crt.der`.

Reply:
I'm afraid `two_directorynames` isn't a valid extension section. Please run:
`touch server5.key && make parse_input/server5-two-directorynames.crt.der`
I got:
```
openssl-3.1.1 req -x509 -outform der -new -subj "/C=UK/O=Mbed TLS/CN=Mbed TLS directoryName SAN" -set_serial 77 -config test-ca.opensslconf -extensions two_directorynames -days 3650 -sha256 -key server5.key -out parse_input/server5-two-directorynames.crt.der
Error checking x509 extension section two_directorynames
40E4A1439B7F0000:error:11000081:X509 V3 routines:do_ext_nconf:unknown extension:crypto/x509/v3_conf.c:92:
40E4A1439B7F0000:error:11000080:X509 V3 routines:X509V3_EXT_nconf_int:error in extension:crypto/x509/v3_conf.c:48:section=two_directorynames, name=O, value=MALFORM_ME
make: *** [Makefile:443: parse_input/server5-two-directorynames.crt.der] Error 1
```

Reply:
https://github.com/Mbed-TLS/mbedtls/commit/9cfea42fb9724feb9ac406d1debbe8ec8f5dcb4b
I verified that the certificate data didn't change after this commit (apart from the random part of the cert), and that the two Makefile recipes that malform this file still work.

Reply:
I think that this is a small change and not worth doing a new PR.
I think that the extension names I used in my new commit suit what happens here better (the second directoryName isn't _bad_, we'll just be malforming it), so it would be best to take my commit over `07391ca`.